pkg/internal/sqlsmith/type.go

```diff
@@ -41,11 +47,19 @@ func (s *Smither) pickAnyType(typ *types.T) *types.T {
 }
 
 func (s *Smither) randScalarType() *types.T {
-	return sqlbase.RandScalarType(s.rnd)
+	var udts []*types.T
+	for _, t := range s.userDefinedTypes {
```

Comment:
These functions are called quite a lot, and it'll be expensive to recompute these slices every call. Precompute them in extractTypes so that it's fast. I'd even precompute both needed ones, i.e. one with udts+SeedTypes and one udts+types.Scalar.

Reply:
Done -- I was thinking about this, but didn't know if it was premature optimization or not.